contracts/external/dao-migrator/src/msg.rs

```diff
@@ -0,0 +1,30 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use dao_core::query::SubDao;
+
+use crate::types::{MigrationParams, V1CodeIds, V2CodeIds};
+
+#[cw_serde]
+pub struct InstantiateMsg {
+    pub migration_params: MigrationParams,
+    pub sub_daos: Option<Vec<SubDao>>,
+    pub v1_code_ids: V1CodeIds,
+    pub v2_code_ids: V2CodeIds,
+}
+
+#[cw_serde]
+pub enum ExecuteMsg {
+    /// Detects the current DAO configuration and performs a migration
+    /// checking state before and after to smoke test the migration's
+    /// success. This module will remove itself on this message's
+    /// completion regardless of the migration's success.
+    MigrateV1ToV2 {
+        sub_daos: Option<Vec<SubDao>>,
+        params: MigrationParams,
+        v1_code_ids: V1CodeIds,
+        v2_code_ids: V2CodeIds,
+    },
+}
```

Comment:
I feel like if we do it, we turn this contract to `dao_migrator_v2` instead of a migrator.
so continue to my previous comment, should this be 0 future thinking?

Reply:
fair point. this is your contract, so we can go with your rules here. :) personally, i tend to optimize robustness > future proof whenever those two are being compared. it's also hard to know if there even will be a DAO DAO v3! i would like v2 to be the last version tbh.